File: core/src/test/java/com/sequenceiq/cloudbreak/converter/AmbariStackDetailsToJsonConverterTest.java
package com.sequenceiq.cloudbreak.converter;

import static org.junit.Assert.assertEquals;

import org.junit.Before;
import org.junit.Test;

import com.sequenceiq.cloudbreak.TestUtil;
import com.sequenceiq.cloudbreak.api.model.AmbariStackDetailsJson;
import com.sequenceiq.cloudbreak.domain.AmbariStackDetails;

public class AmbariStackDetailsToJsonConverterTest extends AbstractEntityConverterTest<AmbariStackDetails> {

    private AmbariStackDetailsToJsonConverter underTest;

    @Before
    public void setUp() {
        underTest = new AmbariStackDetailsToJsonConverter();
    }

    @Test
    public void testConvert() {
        // GIVEN
        // WHEN
        AmbariStackDetailsJson result = underTest.convert(getSource());
        // THEN
        assertEquals("dummyOs", result.getOs());
        assertAllFieldsNotNull(result);
    }

    @Override
    public AmbariStackDetails createSource() {
        return TestUtil.ambariStackDetails();
    }
}

File: core/src/test/java/com/sequenceiq/cloudbreak/converter/AmbariStackDetailsToJsonConverterTest.kt
package com.sequenceiq.cloudbreak.converter

import org.junit.Assert.assertEquals

import org.junit.Before
import org.junit.Test

import com.sequenceiq.cloudbreak.TestUtil
import com.sequenceiq.cloudbreak.domain.AmbariStackDetails
import com.sequenceiq.cloudbreak.api.model.AmbariStackDetailsJson

class AmbariStackDetailsToJsonConverterTest : AbstractEntityConverterTest<AmbariStackDetails>() {

    private var underTest: AmbariStackDetailsToJsonConverter? = null

    @Before
    fun setUp() {
        underTest = AmbariStackDetailsToJsonConverter()
    }

    @Test
    fun testConvert() {
        // GIVEN
        // WHEN
        val result = underTest!!.convert(source)
        // THEN
        assertEquals("dummyOs", result.os)
        assertAllFieldsNotNull(result)
    }

    override fun createSource(): AmbariStackDetails {
        return TestUtil.ambariStackDetails()
    }
}
